// jadelint/target/rules/AddMaxToProgress.js
var AddMaxToProgress, Rule,
  extend = function(child, parent) { for (var key in parent) { if (hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; },
  hasProp = {}.hasOwnProperty;

Rule = require('./../Rule');

AddMaxToProgress = (function(superClass) {
  extend(AddMaxToProgress, superClass);

  function AddMaxToProgress() {
    return AddMaxToProgress.__super__.constructor.apply(this, arguments);
  }

  AddMaxToProgress.prototype.name = 'Add max attribute to progress elements';

  AddMaxToProgress.prototype.level = 'warning';

  AddMaxToProgress.prototype.description = "All `progress` elements must also have a `max` attribute\n\n```jade\n//- Valid\nprogress(value='50' max='100')\n\n//- Invalid\nprogress(value='0.5')\n```";

  AddMaxToProgress.prototype.check = function() {
    var i, len, name, ref, ref1, val;
    if (this.node.type === 'Tag' && this.node.name === 'progress') {
      ref = this.node.attrs;
      for (i = 0, len = ref.length; i < len; i++) {
        ref1 = ref[i], name = ref1.name, val = ref1.val;
        if (name === 'max') {
          return;
        }
      }
      return this.fail();
    }
  };

  return AddMaxToProgress;

})(Rule);

module.exports = AddMaxToProgress;
