// jadelint/target/rules/DontQuoteQTags.js
var DontQuoteQTags, Rule,
  extend = function(child, parent) { for (var key in parent) { if (hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; },
  hasProp = {}.hasOwnProperty;

Rule = require('./../Rule');

DontQuoteQTags = (function(superClass) {
  extend(DontQuoteQTags, superClass);

  function DontQuoteQTags() {
    return DontQuoteQTags.__super__.constructor.apply(this, arguments);
  }

  DontQuoteQTags.prototype.name = 'Dont use quotes in a q tag';

  DontQuoteQTags.prototype.level = 'warning';

  DontQuoteQTags.prototype.description = "Dont use quotes in a `q` tag\n\n```jade\n//- Valid\nq This is a quote\n\n//- Invalid\nq \"This is a quote\"\n```";

  DontQuoteQTags.prototype.check = function() {
    var firstChild, ref, ref1, ref2, subnodes;
    if (this.node.type === 'Tag' && this.node.name === 'q') {
      subnodes = (ref = (ref1 = this.node.block) != null ? ref1.nodes : void 0) != null ? ref : [];
      firstChild = (ref2 = subnodes[0]) != null ? ref2 : {};
      if (firstChild.type === 'Text') {
        if (firstChild.val.trim().indexOf('"') === 0 && firstChild.val.trim().lastIndexOf('"') === firstChild.val.trim().length - 1) {
          return this.fail();
        }
      }
    }
  };

  return DontQuoteQTags;

})(Rule);

module.exports = DontQuoteQTags;
