// jadelint/target/rules/DontUseLegacyEncoding.js
var DontUseLegacyEncoding, Rule,
  extend = function(child, parent) { for (var key in parent) { if (hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; },
  hasProp = {}.hasOwnProperty;

Rule = require('./../Rule');

DontUseLegacyEncoding = (function(superClass) {
  extend(DontUseLegacyEncoding, superClass);

  function DontUseLegacyEncoding() {
    return DontUseLegacyEncoding.__super__.constructor.apply(this, arguments);
  }

  DontUseLegacyEncoding.prototype.name = 'Dont use legacy encoding format';

  DontUseLegacyEncoding.prototype.level = 'error';

  DontUseLegacyEncoding.prototype.description = "Don't use the\n`<meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\">`\nsyntax for specifying charset";

  DontUseLegacyEncoding.prototype.check = function() {
    var i, len, name, ref, ref1, results, val;
    if (this.node.type === 'Tag' && this.node.name === 'meta') {
      ref = this.node.attrs;
      results = [];
      for (i = 0, len = ref.length; i < len; i++) {
        ref1 = ref[i], name = ref1.name, val = ref1.val;
        if (name === 'http-equiv' && this.clean(val) === 'Content-Type') {
          results.push(this.fail());
        } else {
          results.push(void 0);
        }
      }
      return results;
    }
  };

  return DontUseLegacyEncoding;

})(Rule);

module.exports = DontUseLegacyEncoding;
