// jadelint/target/rules/HTMLRootRequiresLang.js
var HTMLRootRequiresLang, Rule,
  extend = function(child, parent) { for (var key in parent) { if (hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; },
  hasProp = {}.hasOwnProperty;

Rule = require('./../Rule');

HTMLRootRequiresLang = (function(superClass) {
  extend(HTMLRootRequiresLang, superClass);

  function HTMLRootRequiresLang() {
    return HTMLRootRequiresLang.__super__.constructor.apply(this, arguments);
  }

  HTMLRootRequiresLang.prototype.name = 'HTML root requires a lang attribute';

  HTMLRootRequiresLang.prototype.level = 'error';

  HTMLRootRequiresLang.prototype.description = "The root `html` element must have a `lang` attribute\n\n```jade\n//- Valid\nhtml(lang='en-US')\nhtml(lang='jp')\n\n//- Invalid\nhtml\n```";

  HTMLRootRequiresLang.prototype.check = function() {
    var i, len, name, ref, ref1, val;
    if (this.node.type === 'Tag' && this.node.name === 'html') {
      ref = this.node.attrs;
      for (i = 0, len = ref.length; i < len; i++) {
        ref1 = ref[i], name = ref1.name, val = ref1.val;
        if (name === 'lang') {
          return;
        }
      }
      return this.fail();
    }
  };

  return HTMLRootRequiresLang;

})(Rule);

module.exports = HTMLRootRequiresLang;
