// jadelint/target/rules/NoXMLTagNamespace.js
var NoXMLTagNamespace, Rule,
  extend = function(child, parent) { for (var key in parent) { if (hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; },
  hasProp = {}.hasOwnProperty;

Rule = require('./../Rule');

NoXMLTagNamespace = (function(superClass) {
  extend(NoXMLTagNamespace, superClass);

  function NoXMLTagNamespace() {
    return NoXMLTagNamespace.__super__.constructor.apply(this, arguments);
  }

  NoXMLTagNamespace.prototype.name = 'No xmlns attribute allowed';

  NoXMLTagNamespace.prototype.level = 'warning';

  NoXMLTagNamespace.prototype.description = "A tag cannot have an `xmlns` attribute";

  NoXMLTagNamespace.prototype.check = function() {
    var i, len, name, ref, ref1, ref2, results, val;
    ref1 = (ref = this.node.attrs) != null ? ref : [];
    results = [];
    for (i = 0, len = ref1.length; i < len; i++) {
      ref2 = ref1[i], name = ref2.name, val = ref2.val;
      if (name === 'xmlns') {
        results.push(this.fail());
      } else {
        results.push(void 0);
      }
    }
    return results;
  };

  return NoXMLTagNamespace;

})(Rule);

module.exports = NoXMLTagNamespace;
