// jadelint/target/rules/OmitTypeForJS.js
var OmitTypeForJS, Rule,
  extend = function(child, parent) { for (var key in parent) { if (hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; },
  hasProp = {}.hasOwnProperty;

Rule = require('./../Rule');

OmitTypeForJS = (function(superClass) {
  extend(OmitTypeForJS, superClass);

  function OmitTypeForJS() {
    return OmitTypeForJS.__super__.constructor.apply(this, arguments);
  }

  OmitTypeForJS.prototype.name = 'Omit type attribute for JS scripts';

  OmitTypeForJS.prototype.level = 'warning';

  OmitTypeForJS.prototype.description = "Don't do `script(type='text/javascript')`";

  OmitTypeForJS.prototype.check = function() {
    var i, len, name, ref, ref1, results, val;
    if (this.node.type === 'Tag' && this.node.name === 'script') {
      ref = this.node.attrs;
      results = [];
      for (i = 0, len = ref.length; i < len; i++) {
        ref1 = ref[i], name = ref1.name, val = ref1.val;
        if (name === 'type' && this.clean(val) === 'text/javascript') {
          results.push(this.fail());
        } else {
          results.push(void 0);
        }
      }
      return results;
    }
  };

  return OmitTypeForJS;

})(Rule);

module.exports = OmitTypeForJS;
