// jadelint/target/rules/UseFigcaptionAtStartOrEnd.js
var Rule, UseFigCaptionAtStartOrEnd,
  extend = function(child, parent) { for (var key in parent) { if (hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; },
  hasProp = {}.hasOwnProperty;

Rule = require('./../Rule');

UseFigCaptionAtStartOrEnd = (function(superClass) {
  extend(UseFigCaptionAtStartOrEnd, superClass);

  function UseFigCaptionAtStartOrEnd() {
    return UseFigCaptionAtStartOrEnd.__super__.constructor.apply(this, arguments);
  }

  UseFigCaptionAtStartOrEnd.prototype.name = 'Only use the figcaption tag at start or end of figure';

  UseFigCaptionAtStartOrEnd.prototype.level = 'warning';

  UseFigCaptionAtStartOrEnd.prototype.description = "Only use the `figcaption` tag at the beginning or end of a `figure` tag\n\n```jade\n//- Valid\nfigure\n  img(src='/img/figure1.png' alt='Some Figure')\n  figcaption Figure 1\n\nfigure\n    figcaption Figure 1\n    img(src='/img/figure1.png' alt='Some Figure')\n\n//- Invalid\nfigure\n    img(src='/img/figure1.png' alt='Some Figure')\n    figcaption Figure 1\n    img(src='/img/figure1.png' alt='Some Figure')\n```";

  UseFigCaptionAtStartOrEnd.prototype.check = function() {
    var i, index, len, ref, ref1, results, subnode, subnodes;
    if (this.node.type === 'Tag' && this.node.name === 'figure') {
      subnodes = (ref = (ref1 = this.node.block) != null ? ref1.nodes : void 0) != null ? ref : [];
      results = [];
      for (index = i = 0, len = subnodes.length; i < len; index = ++i) {
        subnode = subnodes[index];
        if (subnode.name === 'figcaption' && index !== 0 && index !== subnodes.length - 1) {
          results.push(this.fail());
        } else {
          results.push(void 0);
        }
      }
      return results;
    }
  };

  return UseFigCaptionAtStartOrEnd;

})(Rule);

module.exports = UseFigCaptionAtStartOrEnd;
