// jadelint/target/rules/UseTitleForAbbr.js
var Rule, UseTitleForAbbr,
  extend = function(child, parent) { for (var key in parent) { if (hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; },
  hasProp = {}.hasOwnProperty;

Rule = require('./../Rule');

UseTitleForAbbr = (function(superClass) {
  extend(UseTitleForAbbr, superClass);

  function UseTitleForAbbr() {
    return UseTitleForAbbr.__super__.constructor.apply(this, arguments);
  }

  UseTitleForAbbr.prototype.name = 'Use a title attribute on abbr elements';

  UseTitleForAbbr.prototype.level = 'warning';

  UseTitleForAbbr.prototype.description = "Use a `title` attribute for `abbr` tags\n\n```jade\n//- Valid\nabbr(title='War and Peace') WaP\nabbr(title=\"Of Mice and Men\") OMaM\n\n//- Invalid\nabbr OMaM\n```";

  UseTitleForAbbr.prototype.check = function() {
    var i, len, name, ref, ref1, val;
    if (this.node.type === 'Tag' && this.node.name === 'abbr') {
      ref = this.node.attrs;
      for (i = 0, len = ref.length; i < len; i++) {
        ref1 = ref[i], name = ref1.name, val = ref1.val;
        if (name === 'title') {
          return;
        }
      }
      return this.fail();
    }
  };

  return UseTitleForAbbr;

})(Rule);

module.exports = UseTitleForAbbr;
